module-bsp/board/rt1051/bellpx/hal/battery_charger/BatteryCharger.cpp

```diff
@@ -1,86 +1,217 @@
 // Copyright (c) 2017-2021, Mudita Sp. z.o.o. All rights reserved.
 // For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
 
-#include "BatteryCharger.hpp"
+#include "BatteryChargerIRQ.hpp"
+#include "common/WorkerQueue.hpp"
 
-#include <hal/GenericFactory.hpp>
-#include <EventStore.hpp>
-#include <bsp/battery-charger/battery-charger.cpp>
-#include <bsp/fuel_gauge/fuel_gauge.hpp>
+#include "FreeRTOS.h"
+#include "queue.h"
+#include "timers.h"
 
-namespace hal::battery
+#include <hal/battery_charger/AbstractBatteryCharger.hpp>
+#include <devices/power/MP2639B.hpp>
+#include <devices/power/CW2015.hpp>
+#include <board/BoardDefinitions.hpp>
+
+namespace
 {
-    BatteryCharger::BatteryCharger(AbstractBatteryCharger::BatteryChargerEvents &eventsHandler)
-        : eventsHandler(eventsHandler)
-    {}
+    using namespace bsp::devices::power;
+
+    constexpr auto emergency_poll_time = pdMS_TO_TICKS(5 * 60 * 1000); /// 5min
```

Comment:
What exactly is the emergency here? I have a clue, however I believe it will become not obvious quickly. Maybe a really brief comment here would be appropriate.